site/_posts/2022-10-30-spring-and-fusionAuth.md

```diff
@@ -0,0 +1,353 @@
+
+We create a new OAuth request resolver to add [PKCE](https://oauth.net/2/pkce/) support for any OAuth requests.
+
+Then we add a few things to the HTTP request chain:
+
+- A request authentication test, using `authorizeRequests`. This allows public access to the home page at `/` but requires any other request to first be authenticated (`anyRequest().authenticated()`). 
+- OAuth 2.0 Login. This will redirect any requests to the built-in Spring Security `/login` endpoint to the OAuth 2.0 provider configured in the `application.properties` file. We also add the customized PKCE resolver to the login endpoint configuration.
+- A redirect back to the homepage `/` after a successful logout event.
+
+## Adding the home page and controller
+
+Under the `main/java/.../....` folder, create a new file named `HomeController.java`. Add the following contents to this file:
+
+```java
+package com.fusionauth.javafusion;
+
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.RequestMapping;
+
+@Controller
+public class HomeController {
+
+  @Value("${spring.application.name}")
+  String appName;
+
+  @RequestMapping("/")
+  public String homePage(Model model) {
+      model.addAttribute("appName", appName);
+      return "home";
+  }  
+}
+```
+
+Under the `main/java/.../..../resources/templates` folder, create a new file named `home.html`. Add the following contents to the file:
+
+```html
+<html xmlns:th="http://www.w3.org/1999/xhtml" lang="en">
+<head><title>Home Page</title></head>
+<body>
+	<h1>Hello !</h1>
+	<p>Welcome to <span th:text="${appName}">Our App</span></p>
+
+	<p>You can view your profile <a href="/user">here</a></p>
+</body>
+</html>
+```
+
+## Adding the user page and controller
+
+We set the `user-info-uri` property in the `application.properties` file and add the `openid` and `profile` scopes to the `scope` property so that Spring can automatically query FusionAuth for the user's profile and make it available to the app via the [`AuthenticationPrincipal` annotation parameter](https://docs.spring.io/spring-security/site/docs/current/api/org/springframework/security/core/annotation/AuthenticationPrincipal.html). We can use this to build a user controller to show the logged-in user their profile on FusionAuth. You can customize the profile fields stored on FusionAuth and requested during registration. Read more about this [here](https://fusionauth.io/docs/v1/tech/guides/advanced-registration-forms).
+
+Under the `main/java/.../....` folder, create a new file named `UserController.java`. Add the following contents to this file:
+
+```java
+package com.fusionauth.javafusion;
```

Comment:
Can we have the package be `com.fusionauth.example.spring` or something similar, please?

Reply:
Same for other java classes.

Reply:
Updated